api/python/quilt3/registry.py

```diff
@@ -0,0 +1,48 @@
+from datetime import timedelta
+
+import boto3
+from flask import Flask
+from flask_cors import CORS
+from flask_json import as_json, jsonify
+
+app = Flask(__name__)
+app.config['JSON_USE_ENCODE_METHODS'] = True
+app.config['JSON_ADD_STATUS'] = False
+
+sts_client = boto3.client(
+    'sts',
+)
+
+
+
+CORS(app, resources={"/api/*": {"origins": "*", "max_age": timedelta(days=1)}})
+
+@app.route('/api/buckets', methods=['GET'])
+@as_json
+def list_buckets():
+    return dict(
+        buckets=[]
+    )
+
+@app.route('/api/auth/get_credentials', methods=['GET'])
+@as_json
+def get_credentials():
+    """
+    Obtains credentials corresponding to your role.
+
+    Returns a JSON object with three keys:
+        AccessKeyId(string): access key ID
+        SecretKey(string): secret key
+        SessionToken(string): session token
+    """
+    try:
+        creds = sts_client.get_session_token()
```

Comment:
How long are these credentials valid?

Could we potentially create them before launching the catalog, and just pass them to the catalog in the config (or a URL parameter)?

Adding a Flask dependency to the Python client seems a bit crazy.

Reply:
Do you think adding the flash dependency is actually a problem? Flask seems like it adds less than 1MB of dependency (including all subdependencies):
```
(venv) Armands-MacBook-Air:tmp armandmcqueen$ pip install flask --no-cache-dir
Collecting flask
  Downloading https://files.pythonhosted.org/packages/9b/93/628509b8d5dc749656a9641f4caf13540e2cdec85276964ff8f43bbb1d3b/Flask-1.1.1-py2.py3-none-any.whl (94kB)
     |████████████████████████████████| 102kB 984kB/s
Collecting Werkzeug>=0.15
  Downloading https://files.pythonhosted.org/packages/ce/42/3aeda98f96e85fd26180534d36570e4d18108d62ae36f87694b476b83d6f/Werkzeug-0.16.0-py2.py3-none-any.whl (327kB)
     |████████████████████████████████| 327kB 1.4MB/s
Collecting click>=5.1
  Downloading https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl (81kB)
     |████████████████████████████████| 81kB 541kB/s
Collecting itsdangerous>=0.24
  Downloading https://files.pythonhosted.org/packages/76/ae/44b03b253d6fade317f32c24d100b3b35c2239807046a4c953c7b89fa49e/itsdangerous-1.1.0-py2.py3-none-any.whl
Collecting Jinja2>=2.10.1
  Downloading https://files.pythonhosted.org/packages/65/e0/eb35e762802015cab1ccee04e8a277b03f1d8e53da3ec3106882ec42558b/Jinja2-2.10.3-py2.py3-none-any.whl (125kB)
     |████████████████████████████████| 133kB 584kB/s
Collecting MarkupSafe>=0.23
  Downloading https://files.pythonhosted.org/packages/ce/c6/f000f1af136ef74e4a95e33785921c73595c5390403f102e9b231b065b7a/MarkupSafe-1.1.1-cp37-cp37m-macosx_10_6_intel.whl
Installing collected packages: Werkzeug, click, itsdangerous, MarkupSafe, Jinja2, flask
Successfully installed Jinja2-2.10.3 MarkupSafe-1.1.1 Werkzeug-0.16.0 click-7.0 flask-1.1.1 itsdangerous-1.1.0
```
```
-rw-r--r--  1 armandmcqueen  staff    18K Feb 23  2019 MarkupSafe-1.1.1-cp37-cp37m-macosx_10_6_intel.whl
-rw-r--r--  1 armandmcqueen  staff    16K Oct 26  2018 itsdangerous-1.1.0-py2.py3-none-any.whl
```